src/main/java/se/liu/ida/hefquin/engine/queryplan/executable/impl/pullbased/ResultElementIterator.java

```diff
@@ -1,10 +1,16 @@
 package se.liu.ida.hefquin.engine.queryplan.executable.impl.pullbased;
 
 import java.util.Iterator;
+import java.util.NoSuchElementException;
 
 import se.liu.ida.hefquin.engine.data.SolutionMapping;
+import se.liu.ida.hefquin.engine.queryplan.ExecutableOperator;
 
 public interface ResultElementIterator extends Iterator<SolutionMapping>
 {
+    ExecutableOperator getOp();
```

Comment:
Exposing the internal operator here is dangerous. It may lead to bugs if the operator is accidentally used somewhere else.